docs/APP-CONNECTIVITY.md

```diff
@@ -0,0 +1,299 @@
+must consecutively fail few times in a row for the adapter to be deemed disconnected. EVE will then
+switch all network instances with group uplink label that use the "broken" port to another
+matching adapter with working connectivity. The requirement for multiple consecutive test
+passes/failures prevents from uplink flapping, i.e. switching between adapters too often.
+
+Network instance (both Local and Switch) can be configured without any uplink. In this case,
+the network is "air-gapped", meaning that it is not reachable from outside and, likewise,
+it will not provide external connectivity to the applications. Air-gap NIs are used only to connect
+applications running on the same edge device.
+
+### Flow Logging
+
+EVE uses [Linux connection tracking](https://conntrack-tools.netfilter.org/manual.html)
+to periodically (every 2 minutes) record all application TCP and UDP flows. A flow record
+encapsulates application UUID, VIF name, open/close timestamps, src/dst IP/port/proto 5-tuple,
+packet and byte counters.
+
+Additionally, EVE captures DNS packets to make a recording of every DNS request from application.
+This includes the request time, hostname that was being resolved and the returned IP address(es).
+
+A batch of new flow records is published to the controller (POST `/api/v1/edgeDevice/flowlog`)
+inside `FlowMessage`.
+
+### Application IP routing
+
+Most of the edge deployments will deploy applications that will have connectivity to both
+WAN (Internet) and LAN (e.g. shop floor, machine floor).
+There may be a single WAN interface and multiple LAN interfaces. With the IP route
+configurability of (Local) Network instances provided by EVE, we can automatically (zero-touch)
+provision routing for applications that have access to both the Internet and one or more LANs.
+This is possible because EVE allows to:
+
+* use DHCP (option 121) to propagate the *connected routes* to applications (routes for external
+  networks that NI uplinks are connected to)
+* configure a set of *static IP routes* for a network instance and have them propagated
+  to applications
+
+Picture below portrays an example of an application with two interfaces, connected via
+separate network instances to two different network ports. Blue color is used to highlight
+the connected routes, which, when enabled, EVE will automatically propagate into the application.
+
+For external networks one or more routing hops away (i.e. not directly connected to the device),
+user is able to configure static IP routes and let EVE propagate them to the application
+also using DHCP. Static routes are part of the network instance configuration. User configures
+destination IP subnets which the network instance will become the gateway for (from the app
+perspective). In the example below, the propagated static route is highlighted with the green
+color.
+
+![ni-ip-routing](./images/eve-ip-routing.png)
```

Comment:
Wondering if this 'eve-ip-routing' should add one 'default route' in the diagram to one of the external port

Reply:
Added